src/driver.ml

```diff
@@ -65,21 +65,27 @@ end
 module Instrument = struct
   type pos = Before | After
 
-  type t = { transformation :  Parsetree.structure -> Parsetree.structure ; position : pos}
+  type t = { transformation : Expansion_context.Base.t -> Parsetree.structure -> Parsetree.structure ; position : pos}
 
-  let make transformation ~position = { transformation ; position }
+  module V2 = struct
+    let make transformation ~position = { transformation ; position }
+  end
+
+  let make transformation ~position =
+    let transformation _ st = transformation st in
+    V2.make transformation ~position
 end
 
 module Transform = struct
   type t =
     { name            : string
     ; aliases         : string list
-    ; impl            : (Parsetree.structure -> Parsetree.structure) option
-    ; intf            : (Parsetree.signature -> Parsetree.signature) option
+    ; impl            : (Expansion_context.Base.t -> Parsetree.structure -> Parsetree.structure) option
+    ; intf            : (Expansion_context.Base.t -> Parsetree.signature -> Parsetree.signature) option
     ; lint_impl       : (Parsetree.structure -> Lint_error.t list) option
```

Comment:
It's probably worth adding it here as well. What do you think?

Reply:
Good catch!